src/lib.rs

```diff
@@ -118,6 +120,35 @@ impl Library {
         imp::Library::new(filename).map(From::from)
     }
 
+    fn resolve_name<'a>(name: &'a str) -> Cow<'a, str> {
+        match (name.starts_with(DLL_PREFIX), name.ends_with(DLL_SUFFIX)) {
+            (true, true) => Cow::Borrowed(name),
+            (true, false) => Cow::Owned(format!("{}{}", name, DLL_SUFFIX)),
+            (false, true) => Cow::Owned(format!("{}{}", DLL_PREFIX, name)),
+            (false, false) => Cow::Owned(format!("{}{}{}", DLL_PREFIX, name, DLL_SUFFIX)),
+        }
+    }
+
+    /// Loads a library and does resolve it's name to match platform-specific
+    /// naming schemes.
+    ///
+    /// For a given library called `engine`, this method will resolve the name
+    /// to the following:
+    /// - `Linux`: `libengine.so`
+    /// - `macOS`: `libengine.dylib`
+    /// - `Windows`: `engine.dll`
+    ///
+    /// # Note
+    ///
+    /// This function does only work with library names and does not work when
+    /// supplying a path to a library. The function assumes that the library is
+    /// already present inside a path that is detectable and searchable by the
+    /// OS during runtime.
+    pub fn with_name_resolve<T: AsRef<str>>(libname: T) -> Result<Library, Error> {
```

Comment:
Nit: I wouldn’t call formatting a prefix and suffix to a name to form a filename a "resolve", but I’m struggling to come up with anything better right now.

Reply:
Ye, that was more of a last resort name than anything else. I couldn't come up with something better too. The one thing that I had in mind was `raw` that I think that would fit better on the current behavior of `new` than on  this one.